### ISSUE TITLE
Tracking issue: FRI verifier added to the standard library

### ISSUE THREAD

**Issue by grjte** [0xPolygonMiden/miden-vm#548]
### Goal
Implement FRI verifier in Miden `stdlib` such that we can verify a single FRI query in under 1K VM cycles for FRI over domain of $2^{20}$. Ideally, this would be for both degree 2 and 3 extension fields, but for now, we can focus on degree 2 extension field.

### Details

A lot of work has already been done in #425 and #557, but the results we are seeing so far give us about 5K cycles for verifying a single query. We need to investigate the bottlenecks and probably introduce a new instruction (or two) to bring the cycle count down.

```[tasklist]

### Must have

- [x] #599 
- [x] #568
- [ ] #567 
- [x] #591
- [x] #592
- [ ] #564 
```

```[tasklist]

### Nice to have

- [ ] The above for cubic extension of base field.
```

## Working group:
@Al-Kindi-0, @bobbinth, @itzmeanjan

<details>
<summary>Workflow</summary>

- Discussion should happen here or in the related sub-issues.
- PRs should only be merged by the coordinator, to ensure everyone is able to review.
- Aim to complete reviews within 24 hours.
- When a related sub-issue is opened:
    - add it to the list of sub-issues in this tracking issue
- When opening a related PR:
    - request review from everyone in this working group
- When a sub-issue is completed:
    - close the related issue with a comment that links to the PR where the work was completed
</details>

### Coordinator: @Al-Kindi-0

<details>
<summary>Workflow</summary>

The working group coordinator ensures scope & progress tracking are transparent and accurate. They will:

- Merge approved PRs after all working group members have completed their reviews.
    - add the PR # to the relevant section of the current tracking PR.
    - close any completed sub-issue(s) with a comment that links to the PR where the work was completed
- Monitor workflow items and complete anything that slips through the cracks.
- Monitor scope to see if anything is untracked or unclear. Create missing sub-issues or initiate discussion as required.
- Monitor progress to see if there's anything which isn't moving forward. Initiate discussion as required.
- Identify PRs with especially significant changes and add @grjte and @bobbinth for review.
</details>

**Comment by Dominik1999** [comment]
Was there any progress last week? Are you blocked by something?

**Comment by Dominik1999** [comment]
We there some progress last week? If blocked, do you need something to be unblocked?

**Comment by bobbinth** [comment]
Closed by #644, #705, #706, #709, #720 and others.
